lib/puma/server.rb

```diff
@@ -919,13 +938,15 @@ def fast_write(io, str)
       while true
         begin
           n = io.syswrite str
-        rescue Errno::EAGAIN, Errno::EWOULDBLOCK
+        rescue Errno::EAGAIN, Errno::EWOULDBLOCK => e
+          @debug_logger.error_dump(e)
           if !IO.select(nil, [io], nil, WRITE_TIMEOUT)
             raise ConnectionError, "Socket timeout writing data"
           end
 
           retry
-        rescue  Errno::EPIPE, SystemCallError, IOError
+        rescue  Errno::EPIPE, SystemCallError, IOError => e
+          @debug_logger.error_dump(e)
           raise ConnectionError, "Socket timeout writing data"
```

Comment:
Do we rescue this connectionerror somewhere higher up in the stack? You should just be able to error_dump from there instead.

Reply:
Yup, added logging higher up in the stack